src/parser/WebVTT.cpp

```diff
@@ -111,10 +112,8 @@ bool WebVTT::Parse(uint64_t pts, uint32_t duration, const void *buffer, size_t b
           strText = std::string(cbuf, next - cbuf);
           if (!strText.empty() && strText.back() == '\r')
             strText.resize(strText.size() - 1);
-          if (strText.find("&rlm;", 0, 5) == 0)
-            strText.replace(0, 5, "\0xE2\0x80\0xAB");
-          else if (strText.find("&lrm;", 0, 5) == 0)
-            strText.replace(0, 5, "\0xE2\0x80\0xAA");
+          replace_string(strText, "&rlm;", "\xE2\x80\xAB");
```

Comment:
wouldn't it be enough to only remove the else in line 116?

Reply:
no because the replacement position is fixed from 0 to 5, instead there is the need to perform a replace at any point of the string

Reply:
webvtt not always have `&lrm;` at start of string, another example is: `<c.arabic>&lrm;‏‎.‎لست جباناً‎ ؟‎من يقول إنّي جبان‎‏</c.arabic>` so we need to find inside the string

Reply:
ok, fine, the few cpu cycles should not hurt,
Does this fix finally your issues? I don't see any reason to not let it in then.

Reply:
but: do we have multiple lrm in one line? Or even lrm and rml in one single line?
If not, would removing the 5 lead to the same solution?

Reply:
we solve problems with all right to left languages,
i'm waiting a confermation by @Essam311
so if you can review also https://github.com/xbmc/xbmc/pull/17085

Reply:
> but: do we have multiple lrm in one line? Or even lrm and rml in one single line?
> If not, would removing the 5 lead to the same solution?

i have not tried, now i see

Reply:
change done, practically doesn't change much from the original code, but it can be also reused, unfortunately "replace" does not support directly find replace with two strings parameters

Reply:
wait until tomorrow before merge i want do another test

Reply:
no nothing, it's okay